core/src/staked_nodes_updater_service.rs

```diff
@@ -24,112 +22,28 @@ pub struct StakedNodesUpdaterService {
 impl StakedNodesUpdaterService {
     pub fn new(
         exit: Arc<AtomicBool>,
-        cluster_info: Arc<ClusterInfo>,
         bank_forks: Arc<RwLock<BankForks>>,
-        shared_staked_nodes: Arc<RwLock<StakedNodes>>,
-        shared_staked_nodes_overrides: Arc<RwLock<HashMap<Pubkey, u64>>>,
+        staked_nodes: Arc<RwLock<StakedNodes>>,
+        staked_nodes_overrides: Arc<RwLock<HashMap<Pubkey, u64>>>,
     ) -> Self {
         let thread_hdl = Builder::new()
             .name("solStakedNodeUd".to_string())
             .spawn(move || {
-                let mut last_stakes = Instant::now();
                 while !exit.load(Ordering::Relaxed) {
-                    let overrides = shared_staked_nodes_overrides.read().unwrap();
-                    let mut new_id_to_stake = HashMap::new();
-                    let mut total_stake = 0;
-                    let mut max_stake: u64 = 0;
-                    let mut min_stake: u64 = u64::MAX;
-                    if Self::try_refresh_stake_maps(
-                        &mut last_stakes,
-                        &mut new_id_to_stake,
-                        &mut total_stake,
-                        &mut max_stake,
-                        &mut min_stake,
-                        &bank_forks,
-                        &cluster_info,
-                        &overrides,
-                    ) {
-                        let mut shared = shared_staked_nodes.write().unwrap();
-                        shared.total_stake = total_stake;
-                        shared.pubkey_stake_map = new_id_to_stake;
-                    }
+                    let stakes = {
```

Comment:
Just to double check, is 'stakes' from bank.staked_nodes() the same as the original logic of filtering through tvu_peers plus that of its own node?

Reply:
We don't need IP addresses here anymore, and no need to filter through tvu-peers. `tvu_peers` was used initially to populate `ip-addr => stake` mapping: https://github.com/solana-labs/solana/pull/24339